powqutyd/files/src/calculation.c

```diff
@@ -41,8 +41,9 @@ static void *calculation_thread_run(void* param);
  * ===> idx <= (TS_BUFFER_SIZE -32)
  */
 static volatile unsigned int buffer_data_start_idx=0;
-static volatile unsigned int stop_calculation_run = 0, data_ready=0;
-float hw_offset= 0.0, hw_scale=0.0;
+static volatile unsigned int stop_calculation_run = 0;
```

Comment:
I am still questioning the need for volatile here ... as it come with at a performance price

ps: https://blog.regehr.org/archives/28 ... best read for this kind of decisions

Reply:
I have not introduced these variables, but your link states:
"The volatile qualifier forces stores to go to memory and loads to come from memory, giving us a way to ensure visibility across multiple computations (threads, interrupts, coroutines, or whatever)"

stop_calculation_run is set from main thread, while it's (only) read from calculation thread i would assume its safe to keep locks out of the game for this variable.

Reply:
ok ... now with the intention to ensure variable set to 0 across threads I do understand the volantile intention ... lets merge :)